compiler/rustc_target/src/spec/abi.rs

```diff
@@ -94,6 +96,237 @@ pub fn all_names() -> Vec<&'static str> {
     AbiDatas.iter().map(|d| d.name).collect()
 }
 
+pub fn enabled_names(features: &rustc_feature::Features) -> Vec<&'static str> {
+    AbiDatas
+        .iter()
+        .map(|d| d.name)
+        .filter(|name| is_enabled(features, rustc_span::DUMMY_SP, name).is_ok())
+        .collect()
+}
+
+pub enum AbiDisabled {
+    Unstable { feature: Symbol, explain: &'static str },
+    Unrecognized,
+}
+
+macro_rules! gate_feature_post {
```

Comment:
nit: I would change this from a macro to a function, since it no longer affects control flow.

Reply:
The one slight annoyance with that is that `$feature` is used in both `$features.$feature` (bool field for if it's enabled), as well as `sym::$feature`, so the caller would have to duplicate the name of the feature, passing in both. I guess that's okay, though, and probably worth it being a regular function over a macro?

Reply:
Hmm. I agree changing it is a pain. But I also kind of feel that if this is a macro, it's not helping *enough* for it to be worth it - it would be nice to find a way to simplify the giant match some more, e.g. only pass in a list of ABI names and feature gates and have it generate the match for you.

That said, it's a big change unrelated to your improvement, I'm fine with just merging as is.

Reply:
ah! @m-ou-se pointed out that `Features::enabled(Symbol)` exists, that makes it nicer.